test/e2e/service_to_service_test.go

```diff
@@ -274,3 +277,53 @@ func TestServiceToServiceCallViaActivator(t *testing.T) {
 		})
 	}
 }
+
+// This test is similar to TestServiceToServiceCall, but create an external accessible helloworld service instead
+// It verifys that the helloworld service is accessible internally from both internal domain and external domain.
+// But it's only accessible from external via the external domain
+func TestCallToPublicService(t *testing.T) {
```

Comment:
Could we put the E2E test into a separate PR? E2E test may need test-infra reviewers to review.